src/txn/txn.c

```diff
@@ -2043,8 +2043,13 @@ __wt_txn_is_blocking(WT_SESSION_IMPL *session, bool conservative)
      * forced eviction successful. Specifically excuse it if:
      *  * Hasn't done many updates
      *  * Is in the middle of a commit or abort
+     *
+     * This threshold that we're comparing the number of updates to is related and must be greater
+     * than the threshold we use in reconciliation's "need split" helper. If we're going to rollback
+     * a transaction, we need to have considered splitting the page in the case that its updates are
+     * on a single page.
      */
-    if (conservative && (txn->mod_count < 10 || F_ISSET(session, WT_SESSION_RESOLVING_TXN)))
+    if (conservative && (txn->mod_count < 20 || F_ISSET(session, WT_SESSION_RESOLVING_TXN)))
```

Comment:
If there is a relationship here, don't we need to use something other than a hard-coded constant?

Reply:
a615eeb